### PR TITLE
sentinel_one_cloud_funnel: improve dns field handling

### LINKED ISSUE #10071: [SentinelOne Cloud Funnel] Improve DNS question and answer processing
Description
The SentinelOne Cloud Funnel integration currently copies the value from `sentinel_one_cloud_funnel.event.dns.request`.

There was a change to S1 from `<DNS FQDN>` to `type: <type> <DNS FQDN>`.

This should be extracted correctly into the relevant ECS field. E.g. dns.question.name, [dns.question.type](https://www.elastic.co/guide/en/ecs/current/ecs-dns.html#field-dns-question-type)

And the value from `sentinel_one_cloud_funnel.event.dns.response` to `dns.answer.type` (where applicable) and `dns.answer.name`.

Problem
A customer is looking to move to our SentinelOne Cloud Funnel integration in place of their custom ingest pipeline. And they will leverage the ECS DNS question fields for detection.

### PR DESCRIPTION
**Pull request by adrianchen-es**
## Proposed commit message

See title.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Closes #10071

### CODE REVIEW

**Comment by adrianchen-es** [comment]
The sample logs and/or expected logs require updating as well, as discussed with @efd6. 

@efd6, please feel free to scrap this if this is not up to scratch.

File: packages/sentinel_one_cloud_funnel/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.0.1"
```
Comment (efd6) [inline packages/sentinel_one_cloud_funnel/changelog.yml:2]:
```suggestion
- version: "1.1.0"
```

File: packages/sentinel_one_cloud_funnel/data_stream/event/elasticsearch/ingest_pipeline/pipeline-dns.yml [hunk 2]
```diff
@@ -14,14 +14,147 @@ processors:
...
       target_field: sentinel_one_cloud_funnel.event.dns.response
       ignore_missing: true
+  - script:
+      description: |
+        Splits the ctx?.sentinel_one_cloud_funnel?.event?.dns?.response field that contains the DNS answers.
+        Example: "type:  5 f2.taboola.map.fastly.net;::ffff:151.101.66.2;::ffff:151.101.130.2;::ffff:151.101.194.2;::ffff:151.101.2.2;"
+        Process the ctx?.sentinel_one_cloud_funnel?.event?.dns?.request field that contains the DNS question.
+        Example: "type: 1 cloudflare.com"
+      lang: painless
+      if: (ctx?.sentinel_one_cloud_funnel?.event?.dns?.response != null && ctx?.sentinel_one_cloud_funnel?.event?.dns?.response != "") || (ctx?.sentinel_one_cloud_funnel?.event?.dns?.request != null && ctx?.sentinel_one_cloud_funnel?.event?.dns?.request != "")
```
Comment (efd6) [inline packages/sentinel_one_cloud_funnel/data_stream/event/elasticsearch/ingest_pipeline/pipeline-dns.yml:28]:
```suggestion
      if: >-
        (ctx.sentinel_one_cloud_funnel?.event?.dns?.response != null && ctx.sentinel_one_cloud_funnel.event.dns.response != "") ||
        (ctx.sentinel_one_cloud_funnel?.event?.dns?.request != null && ctx.sentinel_one_cloud_funnel.event.dns.request != "")
```

File: packages/sentinel_one_cloud_funnel/data_stream/event/elasticsearch/ingest_pipeline/pipeline-dns.yml [hunk 3]
```diff
@@ -14,14 +14,147 @@ processors:
...
+        "254": "MAILA"
+        "255": "ANY"
+        "65281": "WINS"
+        "65282": "WINSR"
+      source: |-
+        def ips = new ArrayList();
+        def relatedHosts = new ArrayList();
+        def dns = new HashMap();
+
+        if (ctx?.sentinel_one_cloud_funnel?.event?.dns?.request != null && ctx?.sentinel_one_cloud_funnel?.event?.dns?.request != "") {
```
Comment (efd6) [inline packages/sentinel_one_cloud_funnel/data_stream/event/elasticsearch/ingest_pipeline/pipeline-dns.yml:95]:
```suggestion
        if (ctx.sentinel_one_cloud_funnel?.event?.dns?.request != null && ctx.sentinel_one_cloud_funnel.event.dns.request != "") {
```

File: packages/sentinel_one_cloud_funnel/data_stream/event/elasticsearch/ingest_pipeline/pipeline-dns.yml [hunk 4]
```diff
@@ -14,14 +14,147 @@ processors:
...
+            question.put("type", params[parts[1]]);
+            question.put("name", parts[2]);   
+            relatedHosts.add(parts[2]);
+          } else {
+            question.put("name", request); 
+          }
+          dns.question = question;
+        }
+        
+        if (ctx?.sentinel_one_cloud_funnel?.event?.dns?.response != null && ctx?.sentinel_one_cloud_funnel?.event?.dns?.response != "") {
```
Comment (efd6) [inline packages/sentinel_one_cloud_funnel/data_stream/event/elasticsearch/ingest_pipeline/pipeline-dns.yml:110]:
```suggestion
        if (ctx.sentinel_one_cloud_funnel?.event?.dns?.response != null && ctx.sentinel_one_cloud_funnel.event.dns.response != "") {
```

File: packages/sentinel_one_cloud_funnel/manifest.yml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 format_version: "3.0.2"
 name: sentinel_one_cloud_funnel
 title: SentinelOne Cloud Funnel
-version: "1.0.0"
+version: "1.0.1"
```
Comment (efd6) [inline packages/sentinel_one_cloud_funnel/manifest.yml:4]:
```suggestion
version: "1.1.0"
```

**Review by efd6 — COMMENTED** [review]
This needs a test.

**Comment by adrianchen-es** [comment]
@efd6 
2 Questions
- Do you think this PR is aligned with our standards?
   I am happy to close this to avoid further taking up your time.
- The tests would be specific to `test-dns.log` and  `test-dns.log-expected.json`?
   The customer also mentioned the DNS request/response type may not show up depending on the S1 agent's version. 
   So, we'd need at least 2 more test cases? I'm not sure if we would need to mix and match, too. 
     - request and response both include the type
     - request and response without the type

**Comment by efd6** [comment]
Yes, you can add the test cases to `test-dns.log` and then run `elastic-package test pipeline -g` to generate the expected output, validating that it looks OK. Tests covering the two cases would be good, but I don't think we need to do recombinant cases. I'm happy for you to continue with this.

**Review by efd6 — COMMENTED** [review]
nit then LGTM

**Review by efd6 — APPROVED** [review]
Thanks

**Comment by adrianchen-es** [comment]
Thank you very much again @efd6

**Pull request merged**